Tests/Rules/RedundantMemberwiseInitTests.swift

```diff
@@ -902,4 +875,237 @@ class RedundantMemberwiseInitTests: XCTestCase {
         """
         testFormatting(for: input, output, rule: .redundantMemberwiseInit)
     }
+
+    func testDontRemoveRedundantPublicMemberwiseInitFromPublicStruct() {
+        let input = """
+        public struct CardViewAnimationState {
+            public init(
+            style: CardStyle,
+            backgroundColor: UIColor?
+            ) {
+            self.style = style
+            self.backgroundColor = backgroundColor
+            }
+
+            public let style: CardStyle
+            public let backgroundColor: UIColor?
+        }
+        """
+        testFormatting(for: input, rule: .redundantMemberwiseInit, exclude: [.redundantSelf, .trailingSpace, .indent, .wrapArguments])
+    }
```

Comment:
This test case makes sense, however we _do_ want to remove the initializer in this case where the struct is public but the init is internal:

```suggestion
    func testDontRemoveRedundantPublicMemberwiseInitFromPublicStruct() {
        let input = """
        public struct CardViewAnimationState {
            init(
            style: CardStyle,
            backgroundColor: UIColor?
            ) {
            self.style = style
            self.backgroundColor = backgroundColor
            }

            public let style: CardStyle
            public let backgroundColor: UIColor?
        }
        """
        testFormatting(for: input, rule: .redundantMemberwiseInit, exclude: [.redundantSelf, .trailingSpace, .indent, .wrapArguments])
    }
```